### PR TITLE
Fix infinite loop when indenting defined type

### PR DESCRIPTION
**Pull request by darkfeline**
https://github.com/voxpupuli/puppet-mode/issues/77

### CODE REVIEW

**Comment by rski** [comment]
Thanks for the PR, but this doesn't seem to fix the reported issue. I added the code from the issue to the tests and they just hang:

```
~/Code/puppet-mode on  darkfeline-indent-define! ⌚ 10:48:21
$ make test
cask exec ert-runner 
Running tests on Emacs 25.1.1
Loading /home/rski/Code/puppet-mode/puppet-mode.el (source)...

Indenting region...
Indenting region...done

Indenting region...
Indenting region...done

Indenting region...
Indenting region...done

Indenting region...
Indenting region...done

Indenting region...
Indenting region...done

Indenting region...
Indenting region...done

Indenting region...
Indenting region...done

Indenting region...
```

**Comment by darkfeline** [comment]
Attempted proper fix, which is more invasive.

**Comment by darkfeline** [comment]
Could I have this looked at again?

**Comment by rski** [comment]
Hi @darkfeline , sorry I was awol for a bit but im taking a look now :-)

File: puppet-mode.el [hunk 1]
```diff
@@ -544,24 +544,31 @@ of the closing brace of a block."
...
 
+(defun puppet-in-argument-list ()
+  "If point is in an argument list, return the position of the opening '('.
+If point is not in an argument list, return nil."
+  (save-excursion
+    (save-match-data
+      (condition-case nil
+          (progn
+            (backward-up-list)
+            (if (looking-at "(")
```
Comment (darkfeline) [inline puppet-mode.el]: Maybe.  I don't think it's significantly better either way.

**Comment by rski** [comment]
looks good, thank you so much for this @darkfeline and all the other patches. Would you be interested in joining voxpupuli and helping maintain puppet-mode?

**Comment by darkfeline** [comment]
Sorry for the delayed response, I was on vacation.

I would be glad to join, although I cannot promise spending time looking over issues or PRs beyond what's relevant to my work.

**Pull request merged**